WebIDL/ecmascript-binding/global-object-implicit-this-value.any.js

```diff
@@ -0,0 +1,82 @@
+// META: global=window,worker 
+
+// https://heycam.github.io/webidl/#dfn-attribute-getter (step 1.1.2.1)
+// https://heycam.github.io/webidl/#dfn-attribute-setter (step 4.5.1)
+// https://heycam.github.io/webidl/#dfn-create-operation-function (step 2.1.2.1)
+
+const notGlobalObject = Object.create(Object.getPrototypeOf(globalThis));
+
+test(() => {
+    assert_throws_js(TypeError, () => { Object.create(globalThis).self; });
+    assert_throws_js(TypeError, () => { getGlobalPropertyDescriptor("location").get.call(notGlobalObject); });
+    assert_throws_js(TypeError, () => { Reflect.get(globalThis, "navigator", notGlobalObject); });
+    assert_throws_js(TypeError, () => { new Proxy(globalThis, {}).onerror; });
+}, "Global object's getter throws when called on incompatible object");
+
+test(() => {
+    assert_throws_js(TypeError, () => { Object.create(globalThis).origin = origin; });
+    assert_throws_js(TypeError, () => { getGlobalPropertyDescriptor("onerror").set.call(notGlobalObject, onerror); });
+    assert_throws_js(TypeError, () => { Reflect.set(globalThis, "onoffline", onoffline, notGlobalObject); });
+    assert_throws_js(TypeError, () => { new Proxy(globalThis, {}).ononline = ononline; });
+}, "Global object's setter throws when called on incompatible object");
+
+test(() => {
+    assert_throws_js(TypeError, () => { Object.create(globalThis).setTimeout(() => {}, 100); });
+    assert_throws_js(TypeError, () => { clearTimeout.call(notGlobalObject, () => {}); });
+    assert_throws_js(TypeError, () => { Reflect.apply(btoa, notGlobalObject, [""]); });
+    assert_throws_js(TypeError, () => { new Proxy(globalThis, {}).removeEventListener("foo", () => {}); });
+}, "Global object's operation throws when called on incompatible object");
+
+if (typeof document !== "undefined") {
```

Comment:
Why is this guard necessary?

Reply:
Because it's `*.any.js` and workers / worklets have no `document`. My intention was to keep all tests in one file. It indeed looks a bit hacky; would we rather have a separate file for `Window`-only stuff?

Reply:
Ah, right, hmm. Yeah, maybe just pulling this block out into a .window.js file would be a bit easier to understand, but I am OK either way.